### ISSUE TITLE
Defining instances in $HOME/.annis

### ISSUE THREAD

**Issue by dlukes** [korpling/ANNIS#406]
I was trying to define custom instances of the ANNIS interface as per section 6.4 of the User Manual:

> An instance is defined by a JSON file inside the instances sub-folder in one of the
configuration locations, e.g. the home folder of the user running the ANNIS service

But my instance definitions were somehow not being picked up, so I decided to log the config locations which were being searched:

```
17:56:25.016 [http-bio-8080-exec-112] INFO  annis.libgui.AnnisBaseUI - config locations: [/var/lib/tomcat7/webapps/annis-gui/WEB-INF/conf/instances, /etc/annis/instances, /usr/share/tomcat7/.annis/instances] (AnnisBaseUI.java:188)
```

Clearly, it's the home directory of the **user running the ANNIS web interface** (tomcat7 in my case) which is being searched, not that of the user running the ANNIS service (myself). This is the line which would have to be changed in order to get the behavior described in the manual:

https://github.com/korpling/ANNIS/blob/a8f2a928e59202e4b8bf571a95fa00aa022f94d2/annis-libgui/src/main/java/annis/libgui/AnnisBaseUI.java#L184

Though I've no idea whether there's an easy way for the interface to ascertain the user running the service...

**Comment by thomaskrause** [comment]
Hi, it is indeed the users home directory who is running the web-frontend. I asked Amir to change the user guide to reflect it. For users running the Kickstarter this would not make a difference, but for servers it does. 

Since there is not always a possibility to have a home-directory for the "tomcat7" user (and maybe shouldn't be) I would recommend to set the "ANNIS_CFG" environment variable in your startup-script as described in http://korpling.github.io/ANNIS/doc/admin-configure-webapp.html#admin-configure-general

E.g. for the "SNAPSHOT" instance on our korpling-server we have a private tomcat-instance (created with `tomcat7-instance-create`) with a custom "setenv.sh" script containing:

```
export ANNIS_CFG=/etc/annis_snapshot/
```

**Comment by dlukes** [comment]
Thanks for that tip! And for reminding me about the dev/admin guide, it's a place I often forget to check...

I think it might be a good idea to mention this possibility (at least in passing) in section 3.2 of the User Guide, Installing an ANNIS Server. That's generally the point in the setup process where you need to get a hold of your sysadmin for all the configuration that requires `sudo`, and in my experience, once you succeed in usurping your sysadmin's attention, it's better to have all your requests ready in one go :)

**Comment by dlukes** [comment]
Thanks for that tip! And for reminding me about the dev/admin guide, it's a place I often forget to check...

I had no idea about `tomcat7-instance-create`, it could help me a lot in the future -- the less configuration you need to bother your sysadmin with the better :)

**Comment by amir-zeldes** [comment]
The information has been corrected and updated for the upcoming 3.3 user guide (not yet online), which should be released soon.
